src/rust/cryptography-x509-validation/src/policy/mod.rs

```diff
@@ -100,17 +106,70 @@ pub static WEBPKI_PERMITTED_ALGORITHMS: Lazy<HashSet<&AlgorithmIdentifier<'_>>>
     ])
 });
 
+const RFC5280_CRITICAL_CA_EXTENSIONS: &[asn1::ObjectIdentifier] =
+    &[BASIC_CONSTRAINTS_OID, KEY_USAGE_OID];
+const RFC5280_CRITICAL_EE_EXTENSIONS: &[asn1::ObjectIdentifier] =
+    &[BASIC_CONSTRAINTS_OID, SUBJECT_ALTERNATIVE_NAME_OID];
+
 /// Represents a logical certificate "subject," i.e. a principal matching
 /// one of the names listed in a certificate's `subjectAltNames` extension.
 pub enum Subject<'a> {
     DNS(DNSName<'a>),
     IP(IPAddress),
 }
 
+impl Subject<'_> {
```

Comment:
General name matching is likely something that could be extracted and tested in its own PR.

Reply:
https://github.com/pyca/cryptography/pull/9659